numpy/core/tests/test_umath.py

```diff
@@ -740,6 +741,84 @@ def test_float_remainder_corner_cases(self):
                 assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, rem))
 
 
+class TestDivisionOverflows:
+    import operator
+    result_type = namedtuple('result_type',
+            ['nocast', 'casted'])
+    overflow_results = {
+        np.remainder: result_type('0', '0'),
+        np.fmod: result_type('0', '0'),
+        operator.mod: result_type('0', '0'),
+        operator.floordiv: result_type('np.iinfo(dividend_dtype).min',
+            '-np.iinfo(dividend_dtype).min'),
+        np.floor_divide: result_type('np.iinfo(dividend_dtype).min',
+            '-np.iinfo(dividend_dtype).min'),
+        np.divmod: result_type('(np.iinfo(dividend_dtype).min, 0)',
+            '(-np.iinfo(dividend_dtype).min, 0)')
```

Comment:
I would really like to remove the eval.  Could be with a lambda, but I feel there must be a better way.

Reply:
I hope 622ec4c24e362945d4e6850e55a734f43b386147 is ok? Seems a bit overkill, but does the trick.